assets/js/base/components/price-slider/index.js

```diff
@@ -0,0 +1,405 @@
+	 */
+	const getProgressStyle = () => {
+		const low =
+			Math.round( 100 * ( ( value.min - min ) / ( max - min ) ) ) - 0.5;
+		const high =
+			Math.round( 100 * ( ( value.max - min ) / ( max - min ) ) ) + 0.5;
+
+		return {
+			'--low': low + '%',
+			'--high': high + '%',
+		};
+	};
+
+	/**
+	 * Works around an IE issue where only one range selector is visible by changing the display order
+	 * based on the mouse position.
+	 *
+	 * @param {obj} event event data.
+	 */
+	const findClosestRange = ( event ) => {
+		if ( isLoading ) {
+			return;
+		}
+		const bounds = event.target.getBoundingClientRect();
+		const x = event.clientX - bounds.left;
+		const minWidth = minRange.current.offsetWidth;
+		const minValue = minRange.current.value;
+		const maxWidth = maxRange.current.offsetWidth;
+		const maxValue = maxRange.current.value;
+
+		const minX = minWidth * ( minValue / max );
+		const maxX = maxWidth * ( maxValue / max );
+
+		const minXDiff = Math.abs( x - minX );
+		const maxXDiff = Math.abs( x - maxX );
+
+		/**
+		 * The default z-index in the stylesheet as 20. 20 vs 21 is just for determining which range
+		 * slider should be at the front and has no meaning beyond
+		 */
+		if ( minXDiff > maxXDiff ) {
+			minRange.current.style.zIndex = 20;
+			maxRange.current.style.zIndex = 21;
+		} else {
+			minRange.current.style.zIndex = 21;
+			maxRange.current.style.zIndex = 20;
+		}
+	};
+
+	/**
+	 * Called when the slider is dragged.
+	 * @param {obj} event Event object.
+	 */
+	const onDrag = ( event ) => {
+		const isMin = event.target.classList.contains(
+			'wc-block-price-filter__range-input--min'
+		);
+		const values = constrainRangeSliderValues(
+			[ minRange.current.value, maxRange.current.value ],
+			min,
+			max,
+			step,
+			isMin
+		);
+		setValue( {
+			min: parseInt( values[ 0 ], 10 ),
+			max: parseInt( values[ 1 ], 10 ),
+		} );
+	};
+
+	/**
+	 * Called when a price input loses focus.
+	 * @param {obj} event Event object.
+	 */
+	const onInputBlur = ( event ) => {
+		const isMin = event.target.classList.contains(
+			'wc-block-price-filter__amount--min'
+		);
+		const values = constrainRangeSliderValues(
+			[
+				minInput.current.value.replace( /[^0-9.-]+/g, '' ),
+				maxInput.current.value.replace( /[^0-9.-]+/g, '' ),
+			],
+			min,
+			max,
+			step,
+			isMin
+		);
+		setValue( {
+			min: parseInt( values[ 0 ], 10 ),
+			max: parseInt( values[ 1 ], 10 ),
+		} );
+	};
+
+	/**
+	 * Called when the value of a price input is changed.
+	 * @param {obj} event Event object.
+	 */
+	const onInputChange = ( event ) => {
+		const newValue = event.target.value.replace( /[^0-9.-]+/g, '' );
+		const isMin = event.target.classList.contains(
+			'wc-block-price-filter__amount--min'
+		);
+		const key = isMin ? min : max;
+
+		setFormattedValue( {
+			...formattedValue,
+			[ key ]: formatCurrencyForInput(
+				newValue,
+				priceFormat,
+				currencySymbol
+			),
+		} );
+	};
```

Comment:
I realize this might have been the reason why you have formatted value in state.  However, it's possible you could get rid of this event handler entirely if the only reason it's here is for updating the formatted value on input value change.  Even if you do need this, shouldn't it just update the `minPrice` and `maxPrice` values in the local state instead?

With that said _if_ you do need to keep this in (I'd be surprised if you do), then make sure it's wrapped in `useCallback`.